extensions/sql-database-projects/src/models/project.ts

```diff
@@ -676,28 +597,27 @@ export class Project implements ISqlProject {
 			throw new Error(constants.noFileExist(filePath));
 		}
 
-		// Check if file already has been added to sqlproj
 		const normalizedRelativeFilePath = utils.convertSlashesForSqlProj(path.relative(this.projectFolderPath, filePath));
-		const existingEntry = this.files.find(f => f.relativePath.toUpperCase() === normalizedRelativeFilePath.toUpperCase());
-		if (existingEntry) {
-			return existingEntry;
-		}
+		let result: ResultStatus;
 
-		// Ensure that parent folder item exist in the project for the corresponding file path
-		await this.ensureFolderItems(path.relative(this.projectFolderPath, path.dirname(filePath)));
+		if (path.extname(filePath) === constants.sqlFileExtension) {
+			result = await this.sqlProjService.addSqlObjectScript(this.projectFilePath, filePath)
+			await this.readFilesInProject();
+		} else {
+			result = await this.sqlProjService.addNoneItem(this.projectFilePath, filePath);
+			await this.readNoneScripts();
+		}
 
-		// Update sqlproj XML
-		const fileEntry = this.createFileProjectEntry(normalizedRelativeFilePath, EntryType.File);
-		const xmlTag = path.extname(filePath) === constants.sqlFileExtension ? constants.Build : constants.None;
-		await this.addToProjFile(fileEntry, xmlTag);
-		this._files.push(fileEntry);
+		this.throwIfFailed(result);
 
-		return fileEntry;
+		return this.createFileProjectEntry(normalizedRelativeFilePath, EntryType.File);
 	}
 
 	public async exclude(entry: FileProjectEntry): Promise<void> {
-		const toExclude: FileProjectEntry[] = this._files.concat(this._preDeployScripts).concat(this._postDeployScripts).concat(this._noneDeployScripts).concat(this._publishProfiles).filter(x => x.fsUri.fsPath.startsWith(entry.fsUri.fsPath));
-		await this.removeFromProjFile(toExclude);
+		entry.type
+		entry.type
+		//const toExclude: FileProjectEntry[] = this._files.concat(this._preDeployScripts).concat(this._postDeployScripts).concat(this._noneDeployScripts).concat(this._publishProfiles).filter(x => x.fsUri.fsPath.startsWith(entry.fsUri.fsPath));
```

Comment:
this function still needs to be updated?

Reply:
If we go the route of having callers know what entry type they're operating on and calling specific functions in `projec.ts` (like `excludePreDeploymentScript()`, then this function should be deleted entirely.

I'll double-check what's still calling this.